### PR TITLE
Changelog updated for v2.5.0-beta.1

### PR DESCRIPTION
**Pull request by SnailBones**
## Launch Checklist

 - [ ] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+* Added `preclick` event allowing popups to close and open in a new location on one click. ([#10926](https://github.com/mapbox/mapbox-gl-js/pull/10926))
+* Added support for Popup className methods while the popup is not on the map. Direct modifications to the popup container CSS class no longer persist.  ([#10889](https://github.com/mapbox/mapbox-gl-js/pull/10889))
+* Improved collision detection for labels along lines. ([#10918](https://github.com/mapbox/mapbox-gl-js/pull/10918))
+* Added `protected` field to mapbox-api-supported validation. ([#10968](https://github.com/mapbox/mapbox-gl-js/pull/10968))
+
+
+### 🐞 Bug fixes
+
+* Fixed `maxBounds` property not working across the 180th meridian. ([#10903](https://github.com/mapbox/mapbox-gl-js/pull/10903))
+* Fixed `map.getBounds()` returning too-large bounds under some conditions. ([#10909](https://github.com/mapbox/mapbox-gl-js/pull/10909))
```
Comment (karimnaaji) [inline CHANGELOG.md:16]: Should we expand a bit _under some conditions_ to be more clear? IIRC it's when the map is highly pitched and low zoom?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+* Fixed markers not updating position on terrain change. ([#10985](https://github.com/mapbox/mapbox-gl-js/pull/10985))
```
Comment (karimnaaji) [inline CHANGELOG.md:17]: nit: To be bit more specific
```suggestion
* Fixed markers not updating position when toggling terrain on. ([#10985](https://github.com/mapbox/mapbox-gl-js/pull/10985))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+* Fixed gap on edge of map in retina displays. ([#10936](https://github.com/mapbox/mapbox-gl-js/pull/10936))
```
Comment (karimnaaji) [inline CHANGELOG.md:18]: nit: Not 100% sure if this is more accurate
```suggestion
* Fixed gap on edge of map on retina displays. ([#10936](https://github.com/mapbox/mapbox-gl-js/pull/10936))
```

**Comment by karimnaaji** [comment]
Nice first version @SnailBones, left a couple of suggestions and nits.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,23 @@
+## 2.5.0
+
+### Features ✨ and improvements 🏁
+
+* Added queryRenderedFeatures support to heatmap layers. ([#10996](https://github.com/mapbox/mapbox-gl-js/pull/10996))
+* Added support for combined usage of line-gradient and line-dasharray paint properties. ([#10894](https://github.com/mapbox/mapbox-gl-js/pull/10894))
+* Added `preclick` event allowing popups to close and open in a new location on one click. ([#10926](https://github.com/mapbox/mapbox-gl-js/pull/10926))
+* Added support for Popup className methods while the popup is not on the map. Direct modifications to the popup container CSS class no longer persist.  ([#10889](https://github.com/mapbox/mapbox-gl-js/pull/10889))
```
[the remark this answers is not in the document]
Comment (arindam1993) [reply CHANGELOG.md:8]: Might be a good idea to make the seconf sence bolder witha yellow warning symbol so as to attact attention

File: CHANGELOG.md (hunk 1 above)
Comment (SnailBones) [reply CHANGELOG.md:16]: It's three different conditions:
- When the view is looking off a north/south edge of the map and pitch > 0
- When the horizon is visible
-  When terrain is enabled and there is padding.
I was trying to make it more concise, should I just list all of those out?

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (arindam1993) [reply CHANGELOG.md:8]: Similar to this.
Screen Shot 2021-09-15 at 1 14 03 PM

**Review by arindam1993 — COMMENTED** [review]
Just one thing

File: CHANGELOG.md (hunk 1 above)
Comment (karimnaaji) [reply CHANGELOG.md:16]: No that seems a bit much, let's go for what you had then!

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (SnailBones) [reply CHANGELOG.md:8]: Maybe something like this:
* Added support for Popup `addClassName`, `removeClassName` and `toggleClassName` methods while the popup is not on the map.  ([#10889](https://github.com/mapbox/mapbox-gl-js/pull/10889))
  * ⚠️ Note: Direct modifications to the popup container CSS class no longer persist. These methods should be used instead.

**Review by karimnaaji — APPROVED** [review]
Looks good

**Review by arindam1993 — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,23 @@
...
+* Improved collision detection for labels along lines. ([#10918](https://github.com/mapbox/mapbox-gl-js/pull/10918))
+* Added `protected` field to mapbox-api-supported validation. ([#10968](https://github.com/mapbox/mapbox-gl-js/pull/10968))
+
+### 🐞 Bug fixes
+
+* Fixed `maxBounds` property not working across the 180th meridian. ([#10903](https://github.com/mapbox/mapbox-gl-js/pull/10903))
+* Fixed `map.getBounds()` returning too-large bounds under some conditions. ([#10909](https://github.com/mapbox/mapbox-gl-js/pull/10909))
+* Fixed markers not updating position when toggling terrain. ([#10985](https://github.com/mapbox/mapbox-gl-js/pull/10985))
+* Fixed gap on edge of map on retina displays. ([#10936](https://github.com/mapbox/mapbox-gl-js/pull/10936))
+* Fixed SDF images rendering inside text. ([#10989](https://github.com/mapbox/mapbox-gl-js/pull/10989))
```
Comment (rreusser) [inline CHANGELOG.md:19]: Maybe "Correctly ignore and warn about SDF images in formatted text"

Comment (rreusser) [reply CHANGELOG.md:19]: (Retroactive change not required; only just a small note in case someone felt strongly about it.)

**Pull request merged**